frontend/public/components/secrets/create-secret/PullSecretForm.tsx

```diff
@@ -0,0 +1,100 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Dropdown } from '../../utils';
+import {
+  getImageSecretKey,
+  AUTHS_KEY,
+  CreateConfigSubform,
+  UploadConfigSubform,
+  SecretSubFormProps,
+} from '.';
+
+export const PullSecretForm: React.FC<SecretSubFormProps> = ({
+  onChange,
+  onError,
+  onFormDisable,
+  stringData,
+  secretType,
+  isCreate,
+}) => {
+  const [authType, setAuthType] = React.useState('credentials');
+  const { t } = useTranslation();
+
+  const pullSecretData = React.useMemo<PullSecretData>(() => {
+    try {
+      const key = getImageSecretKey(secretType);
+      const jsonContent = stringData[key] ?? '{}';
+      return JSON.parse(jsonContent);
+    } catch (err) {
+      onError(`Error parsing secret's data: ${err.message}`);
+      return {};
+    }
+  }, [stringData, secretType]);
+
+  const onDataChanged = React.useCallback(
+    (secretData: PullSecretData) => {
+      if (!_.isError(secretData)) {
+        onFormDisable(false);
+      }
+      const newDataKey = secretData[AUTHS_KEY] ? '.dockerconfigjson' : '.dockercfg';
+      onChange({
+        stringData: {
+          [newDataKey]: JSON.stringify(secretData),
+        },
+        base64StringData: {},
```

Comment:
Sending an empty object here, is this correct?

Reply:
We don't need to explicitly send an empty object. The onChange handler should handle an undefined value.